### ISSUE TITLE
can I get thread scope variables?

### ISSUE THREAD

**Issue by FraserHutch** [maciejzaleski/JMeter-WebSocketSampler#53]
I need to do something like this for my websocket tests:

- send auth request (system information)
- - this will create a user account keyed on system information
- wait for response (not vital, but would  be nice)
- loop sending sync/initialize requests using same system information

The system information is the app id, a unique UUID. I can randomize that easily with a User Defined Variable via the UUID function, however, user defined variables are apparently calculated once at the start of the test whereas I need them to be generated at thread group scope.Is there a way to do that? If now, how can I achieve this?

In other words, I need a randomly generated UUID for each thread (acting as a user) which is accessible to each sample in the thread group.

Thanks in advance!
